specification/network/resource-manager/Microsoft.Network/2017-09-01/applicationSecurityGroup.json

```diff
@@ -0,0 +1,370 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "title": "NetworkManagementClient",
+    "description": "The Microsoft Azure Network management API provides a RESTful set of web services that interact with Microsoft Azure Networks service to manage your network resources. The API has entities that capture the relationship between an end user and the Microsoft Azure Networks service.",
+    "version": "2017-09-01"
+  },
+  "host": "management.azure.com",
+  "schemes": [
+    "https"
+  ],
+  "consumes": [
+    "application/json",
+    "text/json"
+  ],
+  "produces": [
+    "application/json",
+    "text/json"
+  ],
+  "security": [
+    {
+      "azure_auth": [
+        "user_impersonation"
+      ]
+    }
+  ],
+  "securityDefinitions": {
+    "azure_auth": {
+      "type": "oauth2",
+      "authorizationUrl": "https://login.microsoftonline.com/common/oauth2/authorize",
+      "flow": "implicit",
+      "description": "Azure Active Directory OAuth2 Flow",
+      "scopes": {
+        "user_impersonation": "impersonate your user account"
+      }
+    }
+  },
+  "paths": {
```

Comment:
patch API mising. I know this is standard across NRP, but we should start supporting it atleast for new types.

Reply:
Why do we need Patch if there are no Patch operations on the server-side?

Reply:
By this he meant, why are PATCH operations not supported on the server side?

Reply:
I think that's a broader discussion to be had with the NRP folks, but I think it's a little late for that on this resource. While I agree we never sent Gaurav the swagger after the design meeting, we did go over the operations that would be added for the resource, and I don't recall this concern having been brought up. If it's adding an operation, though, it seems like it could be done later (and not just for the new resources, though NRP folks can confirm @DeepakRajendranMsft), after greater discussion.

Reply:
Wouldn't this also requires changes in NRP itself? It requires new routes there and their implementations. I don't think that's possible for this release.

Reply:
@ravbhatnagar supporting it for some resources (new or old) and not supporting for others is inconsistent behavior IMO.  I feel the right thing to do would be do (in accordance with all RPs)
1. implement this in all resources 
2. Leave at the ARM level.

@amarzavery NRP doesnt support PATCH, i believe compute also doesnt support it.

Reply:
@DeepakRajendranMsft - PATCH API is required as per RPC. If for newer resource types, NRP starts supporting PATCH, the debt will not continue to grow. Something your team should take a call on. Also, as you are aware, NRP has committed to fixing all swagger violations in this semester. So my recommendation would be to see if you can start addressing it for newer types. But its your call in the end.

Reply:
This would require a big change in NRP. I don't think it is worth doing for just one operation for now. These violations should be addressed as a separate feature request.